src/storage/Storage.ts

```diff
@@ -14,11 +14,19 @@ import { Codec } from "@/utils";
 import { Type } from "@/utils";
 import { Hash, BlockHeight, ForkId } from "@/types/types";
 
+type Coordinate = {
+    forkId: ForkId;
+    blockHeight: BlockHeight;
+};
+
+type ExitPoint = Coordinate;
+
 export class Storage {
     private blockStorage = new BlockStorage();
     private joinChannelBlockStorage = new JoinChannelBlockStorage();
     private exitChannelBlockStorage = new ExitChannelBlockStorage();
     private stateSnapshotStorage = new StateSnapshotStorage();
+    private exitPoints: ExitPoint[] = [];
```

Comment:
better to do a mapping of `forkId` -> `blockHeights [ ]` - so when you're constructing the state proof for a given fork you fetch all the milestones/blockHeights for it immediately - like this you would have to iterate and check every time is `forkId == forkId` and collect `blockHeight` if true